### PR TITLE
Drive home the point that Sorbet erases generics

### PR DESCRIPTION
**Pull request by jez**
### Motivation

I've seen this asked multiple times this week.

### Test plan

See included automated tests.
Screen Shot 2023-02-28 at 4 44 02 PM

### CODE REVIEW

File: core/types/calls.cc [hunk 1]
```diff
@@ -1626,12 +1626,30 @@ DispatchResult badMetaTypeCall(const GlobalState &gs, const DispatchArgs &args,
                                const TypePtr &wrapped) {
     if (auto e = gs.beginError(errLoc, errors::Infer::MetaTypeDispatchCall)) {
         e.setHeader("Call to method `{}` on `{}` mistakes a type for a value", args.name.show(gs), wrapped.show(gs));
+
+        if (isa_type<SelfTypeParam>(wrapped)) {
+            auto selfTypeParam = cast_type_nonnull<SelfTypeParam>(wrapped);
+            if (selfTypeParam.definition.isTypeMember()) {
```
Comment (froydnj) [inline core/types/calls.cc:1632]:
```suggestion
        if (auto *selfTypeParam = cast_type<SelfTypeParam>(wrapped)) {
            if (selfTypeParam->definition.isTypeMember()) {
```

File: core/types/calls.cc [hunk 2]
```diff
@@ -1626,12 +1626,30 @@ DispatchResult badMetaTypeCall(const GlobalState &gs, const DispatchArgs &args,
...
+                e.addErrorNote("Sorbet erases all generics, so `{}` is never a meaningful, concrete type at runtime.\n"
+                               "    If you want to call a method on some class, that class object must be an argument "
+                               "to this method.",
+                               selfTypeParam.show(gs));
```
Comment (froydnj) [inline core/types/calls.cc:1636]:
```suggestion
                               selfTypeParam->show(gs));
```

**Review by froydnj — APPROVED** [review]
WFM!

File: core/types/calls.cc (hunk 1 above)
Comment (jez) [reply core/types/calls.cc:1632]: This type is inlined so it didn’t let me write this.

**Pull request merged**